sktime/series_as_features/base/estimators/_ensemble.py

```diff
@@ -0,0 +1,355 @@
+from sklearn.utils.fixes import _joblib_parallel_args
+from sklearn.exceptions import DataConversionWarning
+from sklearn.utils import check_array
+from sklearn.utils import check_random_state
+from sklearn.utils import compute_sample_weight
+from sktime.utils.validation.series_as_features import check_X_y
+
+
+def _parallel_fit_estimator(estimator, X, y, sample_weight=None):
+    """Private function used to fit an estimator within a job."""
+    if sample_weight is not None:
+        try:
+            estimator.fit(X, y, sample_weight=sample_weight)
+        except TypeError as exc:
+            if "unexpected keyword argument 'sample_weight'" in str(exc):
+                raise TypeError(
+                    "Underlying estimator {} does not support sample weights."
+                    .format(estimator.__class__.__name__)
+                ) from exc
+            raise
+    else:
+        estimator.fit(X, y)
+    return estimator
+
+
+def _parallel_build_trees(tree, forest, X, y, sample_weight, tree_idx, n_trees,
+                          verbose=0, class_weight=None,
+                          n_samples_bootstrap=None):
+    """
+    Private function used to fit a single tree in parallel."""
+    if verbose > 1:
+        print("building tree %d of %d" % (tree_idx + 1, n_trees))
+
+    # name of step of final estimator in pipeline
+    estimator = tree.steps[-1][0]
+
+    if forest.bootstrap:
+        n_samples = X.shape[0]
+        if sample_weight is None:
+            curr_sample_weight = np.ones((n_samples,), dtype=np.float64)
+        else:
+            curr_sample_weight = sample_weight.copy()
+
+        indices = _generate_sample_indices(tree.random_state, n_samples,
+                                           n_samples_bootstrap)
+        sample_counts = np.bincount(indices, minlength=n_samples)
+        curr_sample_weight *= sample_counts
+
+        if class_weight == 'subsample':
+            with catch_warnings():
+                simplefilter('ignore', DeprecationWarning)
+                curr_sample_weight *= compute_sample_weight('auto', y, indices)
+        elif class_weight == 'balanced_subsample':
+            curr_sample_weight *= compute_sample_weight('balanced', y, indices)
+        fit_params = {f'{estimator}__sample_weight': curr_sample_weight}
+        tree.fit(X, y, **fit_params)
+    else:
+        fit_params = {f'{estimator}__sample_weight': sample_weight}
+        tree.fit(X, y, **fit_params)
+
+    return tree
+
+
+class BaseTimeSeriesForest(BaseForest):
+    """
+    Base class for forests of trees.
+    """
+
+    @abstractmethod
+    def __init__(self,
+                 base_estimator,
+                 n_estimators=100,
+                 estimator_params=tuple(),
+                 bootstrap=False,
+                 oob_score=False,
+                 n_jobs=None,
+                 random_state=None,
+                 verbose=0,
+                 warm_start=False,
+                 class_weight=None,
```

Comment:
`class_weights` is specific to the classifier, so should be removed here

Reply:
But class_weight is used insidethe `fit` function, where it would be a requirement for the classifier? Since the fit function is common for both classifier and regressor, it should be just `None` for the regressor and hence won't affect the regressor?

Reply:
Okay, let's do that, not ideal, but otherwise it would require more refactoring, please remove `class_weights` from init signature of the regressor